lightningd/gossip_control.c

```diff
@@ -41,7 +41,30 @@ static void peer_bad_message(struct subd *gossip, const u8 *msg)
 	tal_free(peer);
 }
 
-static void peer_nongossip(struct subd *gossip, const u8 *msg, int fd)
+static void peer_failed(struct subd *gossip, const u8 *msg)
+{
+	u64 unique_id;
+	struct peer *peer;
+	u8 *err;
+
+	if (!fromwire_gossipstatus_peer_failed(msg, msg, NULL,
+					       &unique_id, &err))
+		fatal("Gossip gave bad PEER_FAILED message %s",
+		      tal_hex(msg, msg));
+
+	peer = peer_by_unique_id(gossip->ld, unique_id);
+	if (!peer)
+		fatal("Gossip gave bad peerid %"PRIu64, unique_id);
```

Comment:
This will swallow the `msg` if we for some reason lose track of the peer in the main daemon. Maybe we can print the message anyway, despite not being able to attribute it correctly?